core/include/stdout_redirect.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * Part of esp-open-rtos
+ * Copyright (C) 2016 Oto Petrik <oto.petrik@gmail.com>
+ * BSD Licensed as described in the file LICENSE
+ */
+
+#ifndef _STDOUT_REDIRECT_H_
+#define _STDOUT_REDIRECT_H_
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+typedef long _WriteFunction(struct _reent *r, int fd, const char *ptr, int len );
+
+void set_write_stdout(_WriteFunction *f);
+_WriteFunction *get_write_stdout();
```

Comment:
It allows multiple implementations of console redirection libraries.
If the redirection library calls the old function (saved from gettter) at the end of its newly installed one, the libraries can stack.
A library that sends console data over a network and a library that logs them to filesystem can work at the same time (and still display data to serial console)